crates/uv-cache-info/src/glob.rs

```diff
@@ -0,0 +1,293 @@
+use std::{
+    collections::BTreeMap,
+    path::{Component, Components, Path, PathBuf},
+};
+
+fn is_glob_like(part: Component) -> bool {
+    matches!(part, Component::Normal(_))
+        && part
+            .as_os_str()
+            .to_str()
+            .is_some_and(|part| ["*", "{", "}", "?"].into_iter().any(|c| part.contains(c)))
+}
+
+#[derive(Debug, Default, Clone, PartialEq, Eq)]
+struct GlobParts {
+    base: PathBuf,
+    pattern: PathBuf,
+}
+
+/// Split a glob into longest possible base + shortest possible glob pattern.
+fn split_glob(pattern: impl AsRef<str>) -> GlobParts {
+    let pattern: &Path = pattern.as_ref().as_ref();
+
+    let mut glob = GlobParts::default();
+    let mut current = PathBuf::new();
```

Comment:
This might be vestigial? It's pushed to below, but then appears unused.

Reply:
I do wonder if this code can be simplified by doing `let mut current = &mut glob.base;`. And then in the loop below:

```rust
if !globbing && is_glob_like(part) {
    current = &mut glob.pattern;
}
```

Then you can just do `current.push(last);` in both places and avoid the extra case analysis on `globbing`.

(I'm wondering if this is what you had and it didn't work for some reason.)

Reply:
Good catch, this is a leftover from a previous version that wasn't cleaned up after a refactor.